tests/test-pipelineruns/test-pipelinerun-gcs.yaml

```diff
@@ -0,0 +1,177 @@
+                - name: patterns
+                  value: $(params.cachePatterns)
+                - name: source
+                  value: $(params.registry)/cache-go:{{hash}}
+                - name: cachePath
+                  value: $(workspaces.source.path)/cache
+                - name: workingdir
+                  value: $(workspaces.source.path)/repo
+                - name: awsCredentialFile
+                  value: $(workspaces.cred.path)/credentials
+                - name: awsConfigFile
+                  value: $(workspaces.cred.path)/config
+                - name: googleCredentialsPath
+                  value: $(workspaces.cred.path)/creds.json
+
+            - name: run-go-build
+              workingDir: $(workspaces.source.path)/repo
+              image: $(params.image)
+              env:
+                - name: GOCACHE
+                  value: $(workspaces.source.path)/cache/gocache
+                - name: GOMODCACHE
+                  value: $(workspaces.source.path)/cache/gomodcache
+              script: |
+                set -x
+                git config --global --add safe.directory $(workspaces.source.path)/repo
+                $(params.buildCommand)
+                echo "Cache size is $(du -sh $(workspaces.source.path)/cache)"
+            - name: cache-upload
+              ref:
+                name: cache-upload
+              params:
+                - name: patterns
+                  value: $(params.cachePatterns)
+                - name: target
+                  value: $(params.registry)/cache-go:{{hash}}
+                - name: cachePath
+                  value: $(workspaces.source.path)/cache
+                - name: workingdir
+                  value: $(workspaces.source.path)/repo
+                - name: awsCredentialFile
+                  value: $(workspaces.cred.path)/credentials
+                - name: awsConfigFile
+                  value: $(workspaces.cred.path)/config
+                - name: force-cache-upload
+                  value: $(params.force-cache-upload)
+                - name: googleCredentialsPath
+                  value: $(workspaces.cred.path)/creds.json
+            - name: cache-fetch-test
+              ref:
+                name: cache-fetch
+              params:
+                - name: patterns
+                  value: $(params.cachePatterns)
+                - name: source
+                  value: $(params.registry)/cache-go:{{hash}}
+                - name: cachePath
+                  value: $(workspaces.test.path)/cache
+                - name: workingdir
+                  value: $(workspaces.source.path)/repo
+                - name: awsCredentialFile
+                  value: $(workspaces.cred.path)/credentials
+                - name: awsConfigFile
+                  value: $(workspaces.cred.path)/config
+                - name: googleCredentialsPath
+                  value: $(workspaces.cred.path)/creds.json
+            - name: validate-cache
```

Comment:
@PuneetPunamiya See this. In this step, we validate cache. In the previous step, we fetched catch in a different workspace - `test`.